client.lua

```diff
@@ -935,27 +935,34 @@ local function IsWhitelistedWeaponStress(weapon)
     return false
 end
 
-local shootingSleep = 500
-CreateThread(function() -- Shooting
-    while true do
-        local isArmed = IsPedArmed(PlayerPedId(), 7)
-        if LocalPlayer.state.isLoggedIn and isArmed then
-            local ped = PlayerPedId()
-            local weapon = GetSelectedPedWeapon(ped)
-            if weapon ~= `WEAPON_UNARMED` then
+local hasWeapon = false
+
+AddEventHandler('ox_inventory:currentWeapon', function(currentWeapon)
+    if not currentWeapon then hasWeapon = false return end
```

Comment:
is the event not guaranteeing currentWeapon is set? This seems odd to me, but the check also can't hurt.

Reply:
Nope, when removing the weapon it returns a nil